pkg/vmprovider/providers/vsphere/resources/vm.go

```diff
@@ -158,22 +162,27 @@ func (vm *VirtualMachine) SetPowerState(ctx context.Context, desiredPowerState v
 		return err
 	}
 
-	if taskInfo, err := powerTask.WaitForResult(ctx, nil); err != nil {
-		if te, ok := err.(task.Error); ok {
-			// Ignore error if VM was already in desired state.
-			if ips, ok := te.Fault().(*types.InvalidPowerStateFault); ok && ips.ExistingState == ips.RequestedState {
-				return nil
-			}
-		}
-		if taskInfo != nil {
-			err = errors.Wrapf(err, "%s failed", taskInfo.Name)
-		}
+	// The ShutdownGuest method does not return a task.
+	if powerTask == nil {
+		return nil
+	}
 
-		vm.logger.Error(err, "VM change power state task failed", "state", desiredPowerState)
-		return err
+	taskInfo, err := powerTask.WaitForResult(ctx, nil)
+	if err == nil {
+		return nil
 	}
 
-	return nil
+	if te, ok := err.(task.Error); ok {
+		// Ignore error if VM was already in desired state.
+		if ips, ok := te.Fault().(*types.InvalidPowerStateFault); ok && ips.ExistingState == ips.RequestedState {
+			return nil
+		}
+	}
+	if taskInfo != nil {
+		err = errors.Wrapf(err, "%s failed", taskInfo.Name)
+	}
+	vm.logger.Error(err, "VM change power state task failed", "state", desiredPowerState)
```

Comment:
nit: It will be helpful to know what is the power state transition by just looking at the logs instead of having to look 
```suggestion
	vm.logger.Error(err, "VM change power state task failed", "existing-state", existingPowerState, "desired-state", desiredPowerState)
```